sstables/mx/writer.cc

```diff
@@ -8,6 +8,7 @@
 
 #include "sstables/mx/writer.hh"
 #include "sstables/writer.hh"
+#include "sstables/trie/bti_index.hh"
```

Comment:
> Note: for the row index, we insert the same clustering blocks to
> both indexes.

(but not at the same time any more)

> The logic for choosing the size of the blocks
> hasn't been changed in any way.

I see column_index_size_in_kb still defaults to 64, we should consider decreasing it, say to 8. Better to do it outside this series so it can be justified with tests.

Reply:
> I see column_index_size_in_kb still defaults to 64, we should consider decreasing it, say to 8. Better to do it outside this series so it can be justified with tests.

Why 8, though? We can consider decreasing it all the way down to 1.

(Note that 4 kiB is not the lower bound, because the Data ranges described by the index are not page-aligned. The smaller the range, the better the chance that you'll only need to read 1 compressed chunk instead of 2 compressed chunks).

Reply:
Not just reading fewer chunks, also less O(N) reading rows and more O(log N) searching the index.

Reply:
Of course, still lots of O(N) decompression.